Services/Simulation/JavascriptInterpreter.cs

```diff
@@ -100,17 +81,44 @@ public class JavascriptInterpreter : IJavascriptInterpreter
             try
             {
                 Program program;
-                if (programs.ContainsKey(filename))
+                string filename;
+                bool isInStorage = string.Equals(script.Path.Trim(),
+                    SimulationScript.SimulationScriptPath.Storage.ToString(),
```

Comment:
can these be null?

Reply:
The `Path` won't be null.
 
For those scripts on disk, `Path` would be `"scripts" + System.IO.Path.DirectorySeparatorChar`,
for  those scripts in storage, `Path` would be the `"Storage"`.